modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItBuildIndexOneNodeTest.java

```diff
@@ -0,0 +1,334 @@
+
+/** Integration test for testing the building of an index in a single node cluster. */
+public class ItBuildIndexOneNodeTest extends BaseSqlIntegrationTest {
+    private static final String TABLE_NAME = "TEST_TABLE";
+
+    private static final String ZONE_NAME = zoneName(TABLE_NAME);
+
+    private static final String INDEX_NAME = "TEST_INDEX";
+
+    @Override
+    protected int initialNodes() {
+        return 1;
+    }
+
+    @BeforeEach
+    void setup() {
+        setAwaitIndexAvailability(false);
+    }
+
+    @AfterEach
+    void tearDown() {
+        sql("DROP TABLE IF EXISTS " + TABLE_NAME);
+        sql("DROP ZONE IF EXISTS " + ZONE_NAME);
+
+        CLUSTER.runningNodes().forEach(IgniteImpl::stopDroppingMessages);
+    }
+
+    @Test
+    void testRecoveryBuildingIndex() throws Exception {
+        createZoneAndTable(ZONE_NAME, TABLE_NAME, 1, 1);
+
+        insertPersons(TABLE_NAME, new Person(0, "0", 10.0));
+
+        CompletableFuture<Void> awaitBuildIndexReplicaRequest = new CompletableFuture<>();
+
+        node().dropMessages((s, networkMessage) -> {
+            if (networkMessage instanceof BuildIndexReplicaRequest) {
+                awaitBuildIndexReplicaRequest.complete(null);
+
+                return true;
+            }
+
+            return false;
+        });
+
+        createIndex(TABLE_NAME, INDEX_NAME, "ID");
+
+        assertThat(awaitBuildIndexReplicaRequest, willCompleteSuccessfully());
+
+        assertFalse(isIndexAvailable(node(), INDEX_NAME));
+
+        // Let's restart the node.
+        CLUSTER.stopNode(0);
+        CLUSTER.startNode(0);
+
+        awaitIndexBecomeAvailable(node(), INDEX_NAME);
+    }
+
+    @Test
+    void testBuildingIndex() throws Exception {
+        createZoneAndTable(ZONE_NAME, TABLE_NAME, 1, 1);
+
+        insertPersons(TABLE_NAME, new Person(0, "0", 10.0));
+
+        createIndex(TABLE_NAME, INDEX_NAME, "ID");
+
+        awaitIndexBecomeAvailable(node(), INDEX_NAME);
+    }
+
+    @Test
+    void testBuildingIndexAndInsertIntoTableInParallel() throws Exception {
+        AtomicInteger nextPersonId = new AtomicInteger();
+
+        createTableAndInsertManyPeople(nextPersonId);
+
+        CompletableFuture<Void> awaitIndexBecomeAvailableEventAsync = awaitIndexBecomeAvailableEventAsync(node(), INDEX_NAME);
+
+        CompletableFuture<Integer> insertIntoTableFuture = runAsyncWithWaitStartExecution(() -> {
+            int insertionsCount = 0;
+            int batchSize = 10;
+
+            while (!awaitIndexBecomeAvailableEventAsync.isDone()) {
+                insertPersons(TABLE_NAME, createPeopleBatch(nextPersonId, batchSize));
+
+                insertionsCount += batchSize;
+            }
+
+            return insertionsCount;
+        });
+
+        createIndexForSalaryFieldAndWaitBecomeAvailable();
+
+        assertThat(awaitIndexBecomeAvailableEventAsync, willCompleteSuccessfully());
+        assertThat(insertIntoTableFuture, will(greaterThan(0)));
```

Comment:
```suggestion
        assertThat(insertIntoTableFuture, willBe(greaterThan(0)));
```

Reply:
My English grammar is lame, it doesn’t matter to me and I’ll change it, but most likely I won’t remember it.